### ISSUE TITLE
Regression? Xournalpp cannot export what Xournal can

### ISSUE THREAD

**Issue by robobenklein** [xournalpp/xournalpp#2372]
**Affects versions :**
 - OS: Pop OS 20.10
 - (Linux only) Desktop environment: X11
 - Which version of libgtk: 3.24.23
 - Version of Xournal++: 1.0.19
 - Installation method: Flatpak, Appimage, and .deb for focal

**Describe the bug**
Xournalpp cannot export a file that Xournal is capable of doing.

**To Reproduce**

1. Pick up editing a file started in Xournal (xoj), save as xopp and continue editing
2. Attempt to export said file: export hangs and only writes a portion of the output PDF
3. Export the xopp to xoj
4. File can be exported to pdf from Xournal without issue

**Expected behavior**
If Xournal can export the file correctly, then Xournalpp should be able to as well.

**Screenshots of Problem**

Exported PDF ends with blank page and does not contain the rest of the content:
![image](https://user-images.githubusercontent.com/5423266/98425834-ad39d900-2064-11eb-8a0b-884ffba6c0ad.png)

Same file, but exported to xoj, then to PDF from Xournal: contains rest of content correctly.
![image](https://user-images.githubusercontent.com/5423266/98425883-d3f80f80-2064-11eb-9a5b-42f5d3c3f9b0.png)

**Additional context**

The empty page contains pasted (screenshot data) graphics, however the previous page does as well, so it's not like it can't export those.

I tried different install methods to make sure it wasn't just Flatpak.

Terminal output didn't seem to contain anything obvious like a crash or stacktrace.

I can upload the .xopp if needed.

**Comment by Technius** [comment]
Does this occur on earlier versions of Xournal++?

**Comment by robobenklein** [comment]
How far back a release should I try to go?

**Comment by BMicraft** [comment]
I managed to create a document in xournalpp (master from a few weeks ago) that will not export correctly (with parts missing) even in the same version. Also, when I try to select the parts that won't be exported they instantly vanish and the cursor gets bugged out (see gif). Sometimes xournal also crashes after clicking a few times.

This behavior is the same in current master (ce4a13a0d0ca6aeb5456daaef782e0167a19fc1e)

![glitchy selection](https://user-images.githubusercontent.com/12884838/100559651-a3964080-32b3-11eb-8158-a949b8a012bf.gif)

Here is the bugged file, in case it is of use for debugging: [glitchy.zip](https://github.com/xournalpp/xournalpp/files/5613335/glitchy.zip)
The problem occurs on page 12

**Comment by rolandlo** [comment]
Hi! I fixed your file: 
[repaired.zip](https://github.com/xournalpp/xournalpp/files/5613904/repaired.zip)
After extraction and opening the xopp-file it will ask you to replace the pdf background. Just choose the pdf `ue5.pdf` in the same folder again.

There were two strokes with -NaN as width, which I needed to remove:
```xml
<stroke tool="pen" ts="0ll" fn="" color="#3333ccff" width="-nan">57.65254067 42.51000000 56.68000000 23.11620313</stroke>
<stroke tool="pen" ts="0ll" fn="" color="#3333ccff" width="-nan">58.65407686 42.51000000 56.68000000 22.72824161</stroke>
```

This is the second time I saw NaN getting stored as a width for stroke, causing trouble. We also had one case where there were all zeroes for image dimensions. We have to prevent getting such strokes and images.

**Comment by BMicraft** [comment]
> Hi! I fixed your file:
> [repaired.zip](https://github.com/xournalpp/xournalpp/files/5613904/repaired.zip)
> After extraction and opening the xopp-file it will ask you to replace the pdf background. Just choose the pdf `ue5.pdf` in the same folder again.

Thank you, very much appreciated.

**Comment by rolandlo** [comment]
Since the problem for the current file is solved, I will close this one in favor of #2464.
